### PR TITLE
65288_News_CRUD

### PR DESCRIPTION
**Pull request by yasin-buet**
news1
news2
news3
user news index
news_news

### CODE REVIEW

**Comment by asifbuetcse** [comment]
OK.

File: app/src/Controller/Admin/NewsController.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+/**
+ * News Controller
+ *
+ * @property \App\Model\Table\NewsTable $News
+ */
+class NewsController extends AppController
+{
+    public function isAuthorized($user)
+    {
+        if (in_array($this->request->action, ['index', 'view'])) {
```
Comment (hieubm) [inline app/src/Controller/Admin/NewsController.php]: Have you check: whether normal users can access admin/news or not ?

File: app/src/Template/Layout/default.ctp [hunk 2]
```diff
@@ -41,20 +41,28 @@ $cakeDescription = 'CakePHP: the rapid development php framework';
...
-                        <li><a href="#"><?= $this->Html->link(
-                                        __('Add News'),
-                                        ['controller' => 'Teams', 'action' => 'add', '_full' => true]
-                                        ); ?>
-                            </a>
-                        </li>
-                    <?php endif; ?>
+                <?php endif; ?>
+                <?php if ($this->request->session()->read('Auth.User.role') == 'admin') : ?>
+                    <li><a href="#"><?= $this->Html->link(
```
Comment (hieubm) [inline app/src/Template/Layout/default.ctp]: why you have link inside of <a> tag ?

File: app/src/Template/Admin/News/index.ctp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            <?php foreach ($news as $news): ?>
+            <tr>
+                <td><?= $this->Number->format($news->id) ?></td>
+                <td><?= h($news->title) ?></td>
+                <td><?= h($news->body) ?></td>
+                <td><?= h($news->created) ?></td>
+                <td><?= h($news->modified) ?></td>
+                <td class="actions">
+                    <?= $this->Html->link(__('View'), ['action' => 'view', $news->id]) ?>
+                    <?php if ($this->request->session()->read('Auth.User.role') == 'admin'): ?>
```
Comment (hieubm) [inline app/src/Template/Admin/News/index.ctp]: Only admin can access this page, so we don't have to check role at here anymore

File: app/src/Controller/Admin/NewsController.php [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    {
+
+        if ($this->request->session()->read('Auth.User.role') == 'user') {
+            return false;
+        }
+
+        if (in_array($this->request->action, ['index', 'view'])) {
+            return true;
+        }
+        return parent::isAuthorized($user);
```
Comment (hieubm) [inline app/src/Controller/Admin/NewsController.php]: Not-loggined user can access this site
Please test it carefully

Comment (yasin-buet) [inline app/src/Controller/Admin/NewsController.php]: @hieubm 
I have added beforefilter to resolve the problem :)

Comment (hieubm) [inline app/src/Controller/Admin/NewsController.php]: you just remove this `isAuthorized` function, and it will work, don't need beforeFilter

**Pull request merged**